plugins/inputs/statsd/datadog_test.go

```diff
@@ -0,0 +1,472 @@
+				now:            now.Add(11),
+				priority:       priorityNormal,
+				source:         "default-hostname",
+				sourceTypeName: "this is the source",
+				alertType:      eventInfo,
+			},
+		},
+		{
+			name: "event metadata source type",
+			args: args{
+				now:      now.Add(11),
+				message:  "_e{10,9}:test title|test text|s:this is the source",
+				hostname: "default-hostname",
+			},
+			expected: expected{
+				title:          "test title",
+				text:           "test text",
+				now:            now.Add(11),
+				priority:       priorityNormal,
+				source:         "default-hostname",
+				sourceTypeName: "this is the source",
+				alertType:      eventInfo,
+			},
+		},
+		{
+			name: "event metadata source tags",
+			args: args{
+				now:      now.Add(11),
+				message:  "_e{10,9}:test title|test text|#tag1,tag2:test",
+				hostname: "default-hostname",
+			},
+			expected: expected{
+				title:     "test title",
+				text:      "test text",
+				now:       now.Add(11),
+				priority:  priorityNormal,
+				source:    "default-hostname",
+				alertType: eventInfo,
+				checkTags: map[string]string{"tag1": "true", "tag2": "test", "source": "default-hostname"},
+			},
+		},
+		{
+			name: "event metadata multiple",
+			args: args{
+				now:      now.Add(11),
+				message:  "_e{10,9}:test title|test text|t:warning|d:12345|p:low|h:some.host|k:aggKey|s:source test|#tag1,tag2:test",
+				hostname: "default-hostname",
+			},
+			expected: expected{
+				title:          "test title",
+				text:           "test text",
+				now:            now.Add(11),
+				priority:       priorityLow,
+				source:         "some.host",
+				ts:             int64(12345),
+				alertType:      eventWarning,
+				aggregationKey: "aggKey",
+				sourceTypeName: "source test",
+				checkTags:      map[string]string{"aggregation_key": "aggKey", "tag1": "true", "tag2": "test", "source": "some.host"},
+			},
+		},
+	}
+	for i := range tests {
+		t.Run(tests[i].name, func(t *testing.T) {
+			s := NewTestStatsd()
+			acc := &testutil.Accumulator{}
+			s.acc = acc
+			err := s.parseEventMessage(tests[i].args.now, tests[i].args.message, tests[i].args.hostname)
+			require.Nil(t, err)
+			m := acc.Metrics[0]
+			require.Equal(t, tests[i].expected.title, m.Measurement)
+			require.Equal(t, tests[i].expected.text, m.Fields["text"])
+			require.Equal(t, tests[i].expected.now, m.Time)
+			require.Equal(t, tests[i].expected.ts, m.Fields["ts"])
+			require.Equal(t, tests[i].expected.priority, m.Fields["priority"])
+			require.Equal(t, tests[i].expected.source, m.Tags["source"])
+			require.Equal(t, tests[i].expected.alertType, m.Fields["alert_type"])
+			require.Equal(t, tests[i].expected.aggregationKey, m.Tags["aggregation_key"])
+			require.Equal(t, tests[i].expected.sourceTypeName, m.Fields["source_type_name"])
+			if tests[i].expected.checkTags != nil {
+				require.Equal(t, tests[i].expected.checkTags, m.Tags)
+			}
+		})
+	}
+}
+
+func TestEventError(t *testing.T) {
```

Comment:
tabletests

Reply:
This is a lot longer and less readable as a table test, imo.

Reply:
Change it anyway, I would rather have all failing tests report instead of one big test.  I'm also fine with individual tests.